sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -846,27 +846,34 @@ ssize_t coap_opt_add_string(coap_pkt_t *pkt, uint16_t optnum, const char *string
     return write_len;
 }
 
-ssize_t coap_opt_add_uquery(coap_pkt_t *pdu, const char *key, const char *val)
+ssize_t coap_opt_add_uquery(coap_pkt_t *pkt, const char *key, const char *val)
 {
+    return coap_opt_add_uquery2(pkt, key, strlen(key), val, val ? strlen(val) : 0);
+}
+
+ssize_t coap_opt_add_uquery2(coap_pkt_t *pkt, const char *key, size_t key_len,
+                             const char *val, size_t val_len)
+{
+    assert(pkt);
+    assert(key);
+    assert(key_len);
+
```

Comment:
you could add an assert for `val` here too, like
```
assert(!val || (val && val_len))
```

Reply:
better say: you should add this assert, see other comments

Reply:
Using the test from https://github.com/RIOT-OS/RIOT/pull/13022#discussion_r386330125, I think the following would be better:

```C
assert((!val || !val_len) || (val && val_len));
```

Reply:
My turn. :-) I think the assert should be
```
assert(!val_len || (val && val_len))
```
I think `!val && val_len` is a programming error. OTOH `val && !val_len` says to ignore the value, which is odd but valid.